axlearn/experiments/text/gpt/fuji.py

```diff
@@ -1011,7 +1011,6 @@ def make_single_host_config(base_config_name: str) -> SpmdTrainer.Config:
                 return cfg
 
             # Make single-host config
-            make_single_host_config_func = functools.partial(make_single_host_config, config_name)
-            config_map[f"{config_name}-single-host"] = make_single_host_config_func
+            config_map[f"{config_name}-single-host"] = lambda: make_single_host_config(config_name)
```

Comment:
IIUC, this won't work as is due to how closures work in Python?
```suggestion
            config_map[f"{config_name}-single-host"] = lambda config_name=config_name: make_single_host_config(config_name)
```

Reply:
That should work just fine, like how we captured things in the idiomatic way of defining decorators.
Here is a simple test that captures the pattern:
```
def f():
  x = 1
  return lambda: print(x)

g = f()
g()
```
which correctly prints 1 when we called `g()`.

Reply:
Maybe I missed something, but why doesn't it exhibit the same problem as the following code:

```
fns = []
for i in range(3):
    fns.append(lambda: i)
print( fns[0]() ) # Incorrectly prints 2 instead of 0.
```

Reply:
(The code in the PR I mean. It makes sense that the snippet you posted in the comment doesn't have this issue.)